src/ex4/server/routes/api.js

```diff
@@ -1 +1,72 @@
-// Define your endpoints here (this is your "controller file")
\ No newline at end of file
+// Define your endpoints here (this is your "controller file")
+const { Router } = require("express");
+const express = require("express");
+const ItemManager = require("../services/item_manager.js");
+const router = express.Router();
+const item_manager = new ItemManager();
+
+router.route("/").get(async (req, res) => {
+	try {
+		const allItems = await item_manager.getAllItems();
+		if (!allItems) {
+			throw new Error("something went wrong on getAllItems function.");
+		}
+		res.status(200).send(allItems);
+	} catch (err) {
+		res.status(500).send(err.message);
+	}
+});
+
+router.route("/add").post(async (req, res) => {
+	try {
+		const itemsToFetch = await item_manager.validateInput(req.body);
+		if (
+			(itemsToFetch && itemsToFetch.length != 0 && !itemsToFetch.length) ||
+			itemsToFetch.length > 0
+		) {
+			const creationMessage = await item_manager.handlingInput(itemsToFetch);
```

Comment:
Naming best practices - use function names that describe the action and not present progressive naming as if you are doing it right now `handlingInput` -> `handleInput`.

Also this name is more for the client items manager not for the server, here one would expect to see a more service like function naming like `item_manager.createItem`/`s`